cpu/nrf52/radio/nrf802154/nrf802154_radio.c

```diff
@@ -234,7 +234,7 @@ static int _read(ieee802154_dev_t *dev, void *buf, size_t max_size,
              * Intentionally using a different calculation than the one from
              * figure 122 of the v1.1 product specification. This appears to
              * match real world performance better */
-            radio_info->rssi = _hwval_to_ieee802154_dbm(hwlqi) + IEEE802154_RADIO_RSSI_OFFSET;
+            radio_info->rssi = _hwval_to_ieee802154_dbm(hwlqi) - IEEE802154_RADIO_RSSI_OFFSET;
```

Comment:
Same here, can it overflow?

Reply:
if it comes from the hardware, I would prefer to have it clamped. As described above, we never know what might happen in the radio